libs/autotagging-commons/src/main/java/org/opensearch/rule/action/UpdateRuleRequest.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rule.action;
+
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.autotagging.Attribute;
+import org.opensearch.autotagging.FeatureType;
+import org.opensearch.autotagging.RuleValidator;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * A request for update Rule
+ * @opensearch.experimental
+ */
+public class UpdateRuleRequest extends ActionRequest {
+    private final String _id;
+    private final String description;
+    private final Map<Attribute, Set<String>> attributeMap;
+    private final String featureValue;
+    private final FeatureType featureType;
+
+    /**
+     * constructor for UpdateRuleRequest
+     * @param _id - the rule id to update
+     * @param description - the description to update
+     * @param attributeMap - the attribute values to update
+     * @param featureValue - the feature value to update
+     * @param featureType - the feature type for the rule
+     */
+    public UpdateRuleRequest(
+        String _id,
+        String description,
+        Map<Attribute, Set<String>> attributeMap,
+        String featureValue,
+        FeatureType featureType
+    ) {
+        this._id = _id;
+        this.description = description;
+        this.attributeMap = attributeMap;
+        this.featureValue = featureValue;
+        this.featureType = featureType;
+    }
+
+    /**
+     * Constructs a UpdateRuleRequest from a StreamInput for deserialization
+     * @param in - The {@link StreamInput} instance to read from.
+     */
+    public UpdateRuleRequest(StreamInput in) throws IOException {
+        super(in);
+        _id = in.readString();
+        description = in.readOptionalString();
+        featureType = FeatureType.from(in);
+        attributeMap = in.readMap(i -> Attribute.from(i, featureType), i -> new HashSet<>(i.readStringList()));
+        featureValue = in.readOptionalString();
+    }
+
+    @Override
+    public ActionRequestValidationException validate() {
+        RuleValidator validator = new RuleValidator(description, attributeMap, featureValue, null, featureType);
+        validator.validateUpdatingRuleParams();
+        return null;
```

Comment:
Can we wrap the validation errors into `ActionRequestValidationException` instance instead of returning null if there are validation errors.